packages/experimental/src/smart-session/grantPermissions.ts

```diff
@@ -0,0 +1,70 @@
+import { ChainController, ConnectionController, OptionsController } from '@reown/appkit-core'
+import type {
+  SmartSessionGrantPermissionsRequest,
+  SmartSessionGrantPermissionsResponse
+} from './utils/TypeUtils.js'
+import {
+  assertWalletGrantPermissionsResponse,
+  extractAddress,
+  updateRequestSigner,
+  validateRequest
+} from './helper/index.js'
+import { WalletConnectCosigner } from './utils/WalletConnectCosigner.js'
+import { ERROR_MESSAGES } from './schema/index.js'
+
+/**
+ * Grants permissions to a user's wallet.
+ * @param request SmartSessionGrantPermissionsRequest
+ * @returns SmartSessionGrantPermissionsResponse
+ *
+ */
+export async function grantPermissions(
+  request: SmartSessionGrantPermissionsRequest
+): Promise<SmartSessionGrantPermissionsResponse> {
+  validateRequest(request)
+
+  // Retrieve state values
+  const { projectId } = OptionsController.state
+  const { activeCaipAddress } = ChainController.state
+
+  // Ensure the namespace is supported and extract address
+  const address = extractAddress(activeCaipAddress)
+  if (!activeCaipAddress || !address) {
+    throw new Error(ERROR_MESSAGES.INVALID_ADDRESS)
+  }
+
+  // Instantiate WalletConnectCosigner and process permissions
+  const walletConnectCosigner = new WalletConnectCosigner(projectId)
```

Comment:
Change variable names to not contain WalletConnect

Reply:
changed to cosignerService.